### ISSUE TITLE
Update project to Java 8

### ISSUE THREAD

**Issue by codetaylor** [SudoPlayGames/Joise#14]
The future is here.

**Comment by czyzby** [comment]
Note that this might make it incompatible with older Android and GWT versions used by LibGDX, for example. It is generally a good idea to extend the library with Java 8-friendly APIs in a separate module.

**Comment by codetaylor** [comment]
I agree with you, @czyzby , and I think that retaining as much compatibility as possible is paramount. I'm not sure if extending the library with a Java 8 API would add any benefit. In fact, I can't immediately think of a way the library could improve by moving to Java 8, which is making me rethink the move entirely.

The initial urge to move to Java 8 hit me when I wanted to clean up [all the anonymous classes in the Joise constructor](https://github.com/SudoPlayGames/Joise/blob/3c58b5b94e34946071a2c2a6c6a24b0258ab2d2e/src/com/sudoplay/joise/Joise.java#L72) with lambdas. I'm just not seeing any real benefit of doing so, other than code cleanup. I'm leaning toward retaining a Java 6 level of compatibility at this point. What do you think?

**Comment by czyzby** [comment]
Take a look at [Retrolambda](https://github.com/orfjackal/retrolambda) - this just might be the syntax sugar you need, without any compatibility sacrifices. It takes [minimal setup](https://github.com/evant/gradle-retrolambda) with Gradle. I haven't personally used it in any 1.6-compatible libraries, but since it just modifies class files, you should be fine.

**Comment by codetaylor** [comment]
Wow, that looks amazing! Thank you :)

**Comment by codetaylor** [comment]
Moving to Java 8 using Retrolambda to continue support Java 6 would require GWT users to use, at minimum, [GWT 2.8.0](https://github.com/gwtproject/gwt/releases/tag/2.8.0).
